update porting guides

gtema opened the pull request:
##### SUMMARY
With PR #40532 `shade` library was retired and replaced with direct use
of `openstacksdk`. Porting guides and doc about dynamic inventory were
not updated.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
docs

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.5.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/gtema/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]
```


##### ADDITIONAL INFORMATION
This change need to be also merged into the stable-2.6 branch

gtema committed:
update porting guides

With PR #40532 `shade` library was retired and replaced with direct use
of `openstacksdk`. Porting guides and doc about dynamic inventory were
not updated.

Akasurde commented on docs/docsite/rst/porting_guides/porting_guide_2.6.rst:
```diff
@@ -62,7 +62,7 @@ Noteworthy module changes
 
     $ ansible localhost -m file -a 'path=/tmp/lib state=directory'
-
+* Openstack modules are no longer using ``shade`` library. Instead ``openstacksdk`` is used. Since ``openstacksdk`` should be already present as a dependency to ``shade`` no additional actions are required
```
period at end of statement.

gtema replied:
Agree it should have been there from the beginning. My item is not the only one without period, so adding it in other places too

gtema committed:
missing period at the item end

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.6.rst:
```diff
@@ -62,7 +62,7 @@ Noteworthy module changes
     $ ansible localhost -m file -a 'path=/tmp/lib state=directory'
-
+* Openstack modules are no longer using ``shade`` library. Instead ``openstacksdk`` is used. Since ``openstacksdk`` should be already present as a dependency to ``shade`` no additional actions are required.
 
```
Can you also add a line to say if yyou were using `contrib/inventory/openstack.py` they should now use `contrib/inventory/openstack_inventory.py`

gtema replied:
is "Porting custom scripts" section ok for that? Was thinking about that already, but didn't find really fitting section

gundalow replied:
How about a new section for "Dynamic inventory scripts"?

gtema replied:
@gundalow this way?

gtema committed:
add dynamic inventory scripts section

add a section with dynamic inventory scripts section and list openstack
inventory script renaming

Commit:
don't reintroduce typo

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.6.rst:
```diff
@@ -103,3 +104,8 @@ Networking
+Dynamic inventory scripts
+=========================
+
+* contrib/inventory/openstack.py is renamed to contrib/inventory/openstack_inventory.py.
```
```
``contrib/inventory/openstack.py`` is renamed to ``contrib/inventory/openstack_inventory.py``.
```

Commit:
fix format, add detail about openstack_inventory.py
